From 39c79d5d514ff05fa749bda20d72af18f4b95e12 Mon Sep 17 00:00:00 2001
From: Paul Barker <pbarker@konsulko.com>
Date: Tue, 9 Jun 2020 11:22:06 +0100
Subject: [master/dunfell][PATCH] packagegroup-arago-sysvinit-boot: Skip recipe
 if sysvinit is not enabled
To: meta-arago@arago-project.org

This change fixes the following error seen at parse time when the
meta-arago-distro layer is present but sysvinit is not in
DISTRO_FEATURES:

    ERROR: .../meta-arago-distro/recipes-core/packagegroups/packagegroup-arago-sysvinit-boot.bb: Please ensure that your setting of VIRTUAL-RUNTIME_init_manager (sysvinit) matches the entries enabled in DISTRO_FEATURES
    ERROR: Failed to parse recipe: .../meta-arago-distro/recipes-core/packagegroups/packagegroup-arago-sysvinit-boot.bb

Signed-off-by: Paul Barker <pbarker@konsulko.com>
---
 .../packagegroups/packagegroup-arago-sysvinit-boot.bb        | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/meta-arago-distro/recipes-core/packagegroups/packagegroup-arago-sysvinit-boot.bb b/meta-arago-distro/recipes-core/packagegroups/packagegroup-arago-sysvinit-boot.bb
index 2eb2d777..a6e21745 100644
--- a/meta-arago-distro/recipes-core/packagegroups/packagegroup-arago-sysvinit-boot.bb
+++ b/meta-arago-distro/recipes-core/packagegroups/packagegroup-arago-sysvinit-boot.bb
@@ -1,3 +1,8 @@
+python __anonymous() {
+    if not "sysvinit" in d.getVar("DISTRO_FEATURES").split():
+        raise bb.parse.SkipRecipe("sysvinit is not enabled")
+}
+
 require recipes-core/packagegroups/packagegroup-core-boot.bb
 
 PR = "r0"
-- 
2.26.2

